### PR TITLE
fix(pushsync): close done channel with a defer

### PR DESCRIPTION
**Pull request by istae**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2707)

### CODE REVIEW

File: pkg/pushsync/pushsync.go [hunk 1]
```diff
@@ -351,6 +351,7 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 
 	resultChan := make(chan receiptResult, 1)
 	doneChan := make(chan struct{})
+	defer close(doneChan)
```
Comment (acud) [inline pkg/pushsync/pushsync.go:354]: you cannot do this. please see https://go.dev/play/p/YO9nL9sKMuy
the dangling goroutines would panic

**Review by acud — CHANGES_REQUESTED** [review]

File: pkg/pushsync/pushsync.go (hunk 1 above)
Comment (istae) [reply pkg/pushsync/pushsync.go:354]: this is the done channel, it's read only.

Comment (acud) [reply pkg/pushsync/pushsync.go:354]: oh sorry, looked at the wrong line. so we have a deadlock on the testnet without this, right?

**Review by acud — APPROVED** [review]

File: pkg/pushsync/pushsync.go (hunk 1 above)
Comment (istae) [reply pkg/pushsync/pushsync.go:354]: yeah, if pushToClosest returns with error, there is a goroutine leak.

**Review by mrekucci — APPROVED** [review]

**Pull request merged**